ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -372,6 +378,46 @@ private class JdbcLedgerDao(
         )
       }
     }
+
+    // Part 5: insert signatories into the 'contract_signatories' table
+    val namedSignatoriesParams = contracts
+      .flatMap(
+        c =>
+          c.signatories.map(
+            w =>
+              Seq[NamedParameter](
+                "contract_id" -> c.id.coid,
+                "signatory" -> w
+            ))
+      )
+      .toArray
+
+    if (!namedSignatoriesParams.isEmpty) {
+      executeBatchSql(
+        SQL_INSERT_CONTRACT_SIGNATORIES,
+        namedSignatoriesParams
+      )
+    }
+
+    // Part 6: insert observers into the 'contract_observers' table
```

Comment:
Can we execute these ~five~ six queries as a single transaction/roundtrip to the database? I'm also a bit worried by the length of this method.

Reply:
I think it's fine as is right now, thanks.